### PR TITLE
Add GET and DELETE for instant messenger accounts

### PR DESCRIPTION
**Pull request by nicerice**
Use this call to update and delete an instant messenger account on your profile.
See https://dev.xing.com/docs/put/users/me/instant_messaging_accounts/:account and https://dev.xing.com/docs/delete/users/me/instant_messaging_accounts/:account for more information.

### CODE REVIEW

File: XNGAPIClient/XNGAPIClient+ProfileEditing.m [hunk 1]
```diff
@@ -540,4 +540,29 @@ - (void)putUpdateLegalInformation:(NSString *)content
...
+                                               name:(NSString *)name
+                                            success:(void (^)(id JSON))success
+                                            failure:(void (^)(NSError *error))failure {
+  
+    if (!account || !name) {
+        return;
+    }
+    
+    NSString *path = [NSString stringWithFormat:@"v1/users/me/instant_messaging_accounts/%@", account];
+    NSDictionary *parameters = @{@"name" : name};
```
Comment (m0rk) [inline XNGAPIClient/XNGAPIClient+ProfileEditing.m]:
```objc
NSDictionary *parameters = @{@"name": name};
```

Sorry 😐

File: XNGAPIClient/XNGAPIClient+ProfileEditing.m [hunk 2]
```diff
@@ -540,4 +540,29 @@ - (void)putUpdateLegalInformation:(NSString *)content
...
+    [self putJSONPath:path JSONParameters:parameters success:success failure:failure];
+}
+
+- (void)deleteInstantMessengerAccount:(NSString *)account
+                              success:(void (^)(id JSON))success
+                              failure:(void (^)(NSError *error))failure {
+    if (account.length == 0) {
```
Comment (m0rk) [inline XNGAPIClient/XNGAPIClient+ProfileEditing.m]: Why in this case `account.length == 0` instead of `!account`. Any special reason?

Comment (nicerice) [inline XNGAPIClient/XNGAPIClient+ProfileEditing.m]: That is actually an oversight!

**Comment by m0rk** [comment]
If you add some UnitTests and merge master to get the conflicts resolved, I will merge it into master.

**Pull request merged**